src/main/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot3;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+public class MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCode extends Recipe {
+
+    private static final String HTTP_STATUS_FQ = "org.springframework.http.HttpStatus";
+    private static final String HTTP_STATUS_CODE_FQ = "org.springframework.http.HttpStatusCode";
+    private static final String RESPONSE_ENTITY_EXCEPTION_HANDLER_FQ = "org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler";
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate `ResponseEntityExceptionHandler` from HttpStatus to HttpStatusCode";
+    }
+
+    @Override
+    public String getDescription() {
+        return "With Spring 6 `HttpStatus` was replaced by `HttpStatusCode` in most method signatures in the `ResponseEntityExceptionHandler`.";
+    }
+
+
+    private static final MethodMatcher HANDLER_METHOD = new MethodMatcher(
+            "org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler *(..)", true);
+
+    private static final MethodMatcher HTTP_STATUS_VALUE_OF = new MethodMatcher(
+            HTTP_STATUS_FQ + " valueOf(int)", true);
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesMethod<>(HANDLER_METHOD),
```

Comment:
Can we add a test that does not call `super`? I wonder if we match those cases where we merely define a method, not use it.

Reply:
Added a test case that doesn't call super, which initially didn't match, so adjusted that